### PR TITLE
Support scan key and value in meta

### PR DESCRIPTION
**Pull request by darionyaphet**
* Support scan key and value processor.
* fix scan to support key and value .
* add some small improve

### CODE REVIEW

File: src/meta/test/ProcessorTest.cpp [hunk 1]
```diff
@@ -4,7 +4,6 @@
  *  (found in the LICENSE.Apache file in the root directory)
  */
 
-#include "base/Base.h"
```
Comment (dutor) [inline src/meta/test/ProcessorTest.cpp]: Why did you remove this?

**Comment by dutor** [comment]
How is going?

**Comment by darionyaphet** [comment]
> How is going?

I have rebased this one and reopen it later :)  thanks @dutor  🎉

File: src/meta/test/ProcessorTest.cpp (hunk 1 above)
Comment (darionyaphet) [reply src/meta/test/ProcessorTest.cpp]: duplicate included, `base/Base.h` have imported by `src/meta/processors/BaseProcessor.h`
